app/socket-apis/__tests__/complete-round.js

```diff
@@ -0,0 +1,81 @@
+// https://github.com/EnCiv/civil-pursuit/issues/212
+const dturn = require('../../dturn/dturn')
+const completeRound = require('../complete-round')
+const { Discussions, insertStatementId, getStatementIds } = dturn
+
+const userId = '12345678abcdefgh'
+const synuser = { synuser: { id: userId } }
+
+beforeEach(async () => {
+  jest.spyOn(console, 'error').mockImplementation(() => {})
+
+  Discussions['discussion1'] = {
+    ShownStatements: [],
+    ShownGroups: [],
+    Gitems: [],
+    Uitems: {},
+    group_size: 10,
+    updateUInfo: jest.fn(),
+  }
+})
+
+afterEach(() => {
+  console.error.mockRestore()
+  jest.restoreAllMocks()
+})
+
+// Test 1: User not logged in
+test('Return undefined if user is not logged in.', async () => {
+  const cb = jest.fn()
+  await completeRound.call({}, 'discussion1', 1, [{ id1: 1 }], cb)
+
+  expect(cb).toHaveBeenCalledTimes(1)
+  expect(cb).toHaveBeenCalledWith(undefined)
+  expect(console.error).toHaveBeenCalledWith('Cannot complete round - user is not logged in.')
+})
+
+// Test 2: Discussion not loaded (getStatementIds throws an error)
+test('Return undefined if discussion is not loaded (getStatementIds fails).', async () => {
+  const cb = jest.fn()
+
+  // Use jest.spyOn to mock getStatementIds function, making it return a rejected Promise
```

Comment:
Please use the real getStatementIds.  I know it's more work, but we need to test dturn and our integration with it,  plus as you go though trying to implement it you may have useful questions or find something.